File: Book Examples/ViewModelDemo/app/src/main/java/com/ebookfrenzy/viewmodeldemo/ui/main/MainViewModel.java
package com.ebookfrenzy.viewmodeldemo.ui.main;

import androidx.lifecycle.ViewModel;

public class MainViewModel extends ViewModel {

    private static final Float usd_to_eu_rate = 0.74F;
    private String dollarText = "";
    private Float result = 0F;

    public void setAmount(String value) {
        this.dollarText = value;
        result = Float.valueOf(dollarText)*usd_to_eu_rate;
    }

    public Float getResult()
    {
        return result;
    }

}

File: KotlinGiraffeSamples/ViewModelDemo/app/src/main/java/com/ebookfrenzy/viewmodeldemo/MainViewModel.kt
package com.ebookfrenzy.viewmodeldemo

import androidx.lifecycle.ViewModel

class MainViewModel : ViewModel() {

    private val rate = 0.74f
    private var dollarText = ""
    private var result: Float = 0f

    fun setAmount(value: String) {
        this.dollarText = value
        result = value.toFloat() * rate
    }

    fun getResult(): Float {
        return result
    }
}